identity-account-storage/src/storage/stronghold.rs

```diff
@@ -82,132 +101,226 @@ impl Stronghold {
 #[cfg_attr(not(feature = "send-sync-storage"), async_trait(?Send))]
 #[cfg_attr(feature = "send-sync-storage", async_trait)]
 impl Storage for Stronghold {
-  async fn flush_changes(&self) -> Result<()> {
-    self.snapshot.save().await?;
-    Ok(())
-  }
+  async fn did_create(
+    &self,
+    network: NetworkName,
+    fragment: &str,
+    private_key: Option<PrivateKey>,
+  ) -> Result<(IotaDID, KeyLocation)> {
+    // TODO: Temporarily implemented via in-memory generation as opposed to using stronghold procedures.
+    let keypair: KeyPair = match private_key {
+      Some(private_key) => KeyPair::try_from_private_key_bytes(KeyType::Ed25519, private_key.as_ref())?,
+      None => KeyPair::new(KeyType::Ed25519)?,
+    };
```

Comment:
TODO? Are we fine shipping this?

Reply:
It's either generate the KeyPair here or for every storage in the account. This way, only stronghold is temporary which is where the actual problem comes from. If we don't want to ship this, we need to migrate to the new stronghold first.